src/controllers/connectionManager.ts

```diff
@@ -872,9 +872,22 @@ export default class ConnectionManager {
 
 				// Note: must call flavor changed before connecting, or the timer showing an animation doesn't occur
 				if (this.statusView) {
-					this.statusView.languageFlavorChanged(fileUri, Constants.mssqlProviderName);
+					const configuration = vscode.workspace.getConfiguration(Constants.extensionConfigSectionName);
+					const intelliSenseConfig = configuration.get<IntelliSenseConfig>(Constants.configIntelliSense);
+					const flavor = intelliSenseConfig.defaultQueryEditorLanguageToNone ? Constants.noneProviderName : Constants.mssqlProviderName;
+					this.statusView.languageFlavorChanged(fileUri, flavor);
 					this.statusView.connecting(fileUri, connectionCreds);
-					this.statusView.languageFlavorChanged(fileUri, Constants.mssqlProviderName);
+					this.statusView.languageFlavorChanged(fileUri, flavor);
+
+					// Notify the language service that the editor with the specified URI doesn't have a language flavor to avoid error squiggles
+					if (flavor === Constants.noneProviderName) {
+						SqlToolsServerClient.instance.sendNotification(LanguageServiceContracts.LanguageFlavorChangedNotification.type,
```

Comment:
Does the language service already support this, or does there need to be work over here, too?

Reply:
The language service already supports it..